torch/_functorch/autograd_function.py

```diff
@@ -686,21 +686,35 @@ class AutogradFunctionApply(HigherOrderOperator):
     def __init__(self):
         super().__init__("autograd_function_apply")
 
-    def __call__(self, fwd, bwd, *fwd_args):
+    def __call__(self, fwd, bwd, *fwd_args, **fwd_kwargs):
         saved_values = None
+        args_tensor_mask = fwd_kwargs["args_tensor_mask"]
+        length_of_tensor_args = sum(args_tensor_mask)
+        # Filter out the original tensor args from fwd_args,
+        # lifted freevars should not be args of ApplyTemplate.apply
+        # since we don't need to calculate the gradients of them.
+        new_fwd_args = fwd_args[:length_of_tensor_args]
 
         class ApplyTemplate(torch.autograd.Function):
             @staticmethod
             def forward(ctx, *args):
                 nonlocal saved_values
-                output, saved_values = fwd(None, *args)
+                output, saved_values = fwd(None, *fwd_args)
```

Comment:
I'm not sure this is going to work (but if it does, then go for it). I'm worried that autograd.Function might require that the `args` that get passed in actually be used, so something like:
```
class ApplyTemplate
  @staticmethod
  def forward(ctx, *args):
        actual_args = (*args, *additional_args)
        output, saved_values = fwd(None, *actual_args)
```

Reply:
I think it's the same as what I did: ```fwd_args``` = ```args``` + lifted freevars(which is ```additional_args``` in your example).

Reply:
Not exactly -- the code in this PR looks like:
```
class ApplyTemplate(torch.autograd.Function):
    @staticmethod
     def forward(ctx, *args):
         nonlocal saved_values
         output, saved_values = fwd(None, *fwd_args)
         return output
```
note that `args` isn't actually passed directly to fwd: we pass in `fwd_args` to `fwd`. 

`fwd_args` contains the tensors in `args`, but... I'm not sure if that works. Again, if the tests pass then I'm happy with it.